server/src/internalClusterTest/java/org/elasticsearch/action/admin/indices/create/AutoCreateSystemIndexIT.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.indices.create;
+
+import org.elasticsearch.action.admin.indices.get.GetIndexResponse;
+import org.elasticsearch.common.util.CollectionUtils;
+import org.elasticsearch.indices.TestSystemIndexPlugin;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.util.Collection;
+
+import static org.elasticsearch.indices.TestSystemIndexDescriptor.INDEX_NAME;
+import static org.elasticsearch.indices.TestSystemIndexDescriptor.PRIMARY_INDEX_NAME;
+import static org.hamcrest.Matchers.is;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class AutoCreateSystemIndexIT extends ESIntegTestCase {
```

Comment:
Do we already have a test for the "The named index is actually the alias for a system index" case we discussed earlier? If not, we should add one here.

Reply:
Yes, it's `CreateSystemIndicesIT#testSystemIndexIsAutoCreatedViaAlias`. I made my own test class because I didn't see how to adjust the `CreateSystemIndicesIT#doCreateTest` method to fit what I wanted to test, but I should give it another shot, or at least move these tests to that class so that we don't have to wonder why these tests are split over the two classes awkwardly.